torchtune/training/checkpointing/_checkpointer.py

```diff
@@ -657,6 +657,13 @@ def load_checkpoint(self) -> Dict[str, Any]:
             converted_state_dict[training.MODEL_KEY] = t5_encoder_hf_to_tune(
                 merged_state_dict,
             )
+        elif self._model_type == ModelType.LLAMA4:
+            from torchtune.models.llama4._convert_weights import llama4_hf_to_tune
+
+            text_config = self._config.get("text_config")
```

Comment:
Why pull out text config here?

Reply:
agh we needed this if we were doing permutes or stuff that depended on num_heads etc. We don't need it now, will remove

Reply:
Why don't we need to permute back to HF?

Reply:
Discussed offline, but see [here](https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama4/convert_llama4_weights_to_hf.py#L390-L391)